### DISCUSSION TITLE
AllowedInclude will be support the other aggregate functions?

### DISCUSSION THREAD

**Discussion by biiiiiigmonster** [spatie/laravel-query-builder#591] [Ideas]
hi  @AlexVanderbist
laravel support the other aggregate functions in the near version
like some `withMin` `withMax` `withSum`

AllowedInclude will be support these?

**Comment by freekmurze** [comment]
This is coming in the next major version (v7) of the package.

You will be able to use aggregate includes like this:

```php
QueryBuilder::for(User::class)
    ->allowedIncludes([
        AllowedInclude::min("postsViewsMin", "posts", "views"),
        AllowedInclude::max("postsViewsMax", "posts", "views"),
        AllowedInclude::sum("postsViewsSum", "posts", "views"),
        AllowedInclude::avg("postsViewsAvg", "posts", "views"),
    ]);
```

The suffixes (`Min`, `Max`, `Sum`, `Avg`) are configurable in the `query-builder` config file.

Docs will be available at https://spatie.be/docs/laravel-query-builder/v7/features/including-relationships once v7 is released.
